### DISCUSSION TITLE
Best practice for updating store

### DISCUSSION THREAD

**Discussion by stephanepx** [pmndrs/zustand#1271] [Q&A]
Hi ! 
I'm wondering what's the best practice to trigger an update in a component between this two ways : 

1. 
```typescript
import { useStore } from './index.ts'

const triggerMyAction = () => {
  const prevValue = useStore.getState().value
  useStore.getState().increaseBy(prevValue + 2)
}

const MyComponent = () => {
   return <button onClick={triggerMyAction}>yolo</button>
}
```

2. 
```typescript
import { useStore } from './index.ts'

const MyComponent = () => {
   const [increaseBy, prevValue] = useStore(state => [state.increaseBy, state.value])
   return <button onClick={() => increaseBy(prevValue + 2)}>yolo</button>
}
```

Here I don't need to use value in my component, so to prevent rerenders, is it better to use the first option ?

Thank's for you help !

**Comment by dai-shi** [comment]
Both work, but mean different.
1 reads prevValue when callback is invoked, while 2 reads prevValue when the component renders.

Notice, you need `shallowEqual` for 2 to avoid extra re-render.

---
I would do this:
```js
const useMyStore = create((set) => ({
  value: 0,
  increase: (by) => set((state) => ({ value: state.value + by })),
}))

const MyComponent = () => {
   return <button onClick={() => useMyStore.getState().increase(2)}>yolo</button>
}
```

Or, even: https://github.com/pmndrs/zustand/blob/main/docs/guides/practice-with-no-store-actions.md

**Reply by stephanepx** [reply]
Awesome ! Thank's for your answer !
